### PR TITLE
Fixes to vita and switch input

### LINKED ISSUE #329: cant control map vita
once in game, my right analog stick doesnt work. The map does move sometimes and when it does, it fully goes to one side.

### PR DESCRIPTION
**Pull request by crudelios**
Attempt to fix the problems to PSVita input reported on gitter by @EduVasquezM_twitter.

Also attempts to fix the mouse position and map edge scrolling.

Also fixes the same problems on Switch.

Fix #329.

### CODE REVIEW

File: src/platform/vita/vita_input.c [hunk 1]
```diff
@@ -228,58 +224,49 @@ void vita_handle_analog_sticks(void)
...
-            right = 1;
-        }
-    } else if (right_y <= 0 && right_x <= 0) {
-        // lower left quadrant
-        if ((-right_y) > slope * (-right_x)) {
-            down = 1;
-        }
-        if ((-right_x) > slope * (-right_y)) {
-            left = 1;
+    if ((right_x * right_x + right_y * right_y) > right_joy_dead_zone_squared) {
```
Comment (bvschaik) [inline src/platform/vita/vita_input.c]: No parentheses necessary around the left-hand side

File: src/platform/switch/switch_input.c [hunk 2]
```diff
@@ -261,58 +258,49 @@ void switch_handle_analog_sticks(void)
...
-            right = 1;
-        }
-    } else if (right_y <= 0 && right_x <= 0) {
-        // lower left quadrant
-        if ((-right_y) > slope * (-right_x)) {
-            down = 1;
-        }
-        if ((-right_x) > slope * (-right_y)) {
-            left = 1;
+    if ((right_x * right_x + right_y * right_y) > right_joy_dead_zone_squared) {
```
Comment (bvschaik) [inline src/platform/switch/switch_input.c]: Parentheses are unnecessary around left-hand side

File: src/platform/switch/switch_input.c [hunk 3]
```diff
@@ -261,58 +258,49 @@ void switch_handle_analog_sticks(void)
...
+            // upper right quadrant
+            if (right_y > slope *right_x) {
+                direction_states[ANALOG_UP] = 1;
+            }
+            if (right_x > slope *right_y) {
+                direction_states[ANALOG_RIGHT] = 1;
+            }
+        } else if (right_y > 0 && right_x <= 0) {
+            // upper left quadrant
+            if (right_y > slope *(-right_x)) {
```
Comment (bvschaik) [inline src/platform/switch/switch_input.c]: Unnecessary parentheses around `-right_x`
Comment also valid for the following `if` statements

**Comment by crudelios** [comment]
@devnoname120, before this is merged can you please test?

**Pull request merged**